resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -70,8 +70,17 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
     // If/when feature steps are executed in client mode, they should instead take care of this,
     // and this code should be removed.
     if (!sc.conf.contains(KUBERNETES_EXECUTOR_POD_NAME_PREFIX)) {
-      sc.conf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX,
-        KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name")))
+      val podNamePrefix = KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name"))
+      if (KubernetesUtils.isValidExecutorPodNamePrefix(podNamePrefix)) {
+        sc.conf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX, podNamePrefix)
+      } else {
```

Comment:
The `isValidExecutorPodNamePrefix` can be false in two cases:
1. length is too long
2. does not matches to the `podConfValidator` pattern

The `else` body won't help on the 2. case.
I suggest to check this 2 conditions separately and only slice the string when the length is the only problem.

Reply:
thank you @attilapiros , I see you worried. Seems `KubernetesConf.getResourceNamePrefix` has already formatted the name, so here I assumed it's only can be over length. Do you still think we should split the check into 2 parts ?

Reply:
In that case not necessarily but please add a comment mentioning this.

Reply:
agree, will add the comment.